File: c#Code/Day1.cs
using System;

class Program
{
    static void TesT()
    {
        int[] array1 = { 1, 3, 5, 7, 9, 0, 0, 0, 0, 0 };  // first array with enough space to hold second array
        int[] array2 = { 2, 4, 6, 8, 10 };
        int m = 5;  // initial number of elements in array1
        int n = array2.Length;  // number of elements in array2

        Merge(array1, m, array2, n);

        foreach (int i in array1)
        {
            Console.Write(i + " ");  // prints 1 2 3 4 5 6 7 8 9 10
        }
    }

    static void Merge(int[] nums1, int m, int[] nums2, int n)
    {
        int i = m - 1, j = n - 1, k = m + n - 1;

        while (i >= 0 && j >= 0)
        {
            if (nums1[i] > nums2[j])
            {
                nums1[k--] = nums1[i--];
            }
            else
            {
                nums1[k--] = nums2[j--];
            }
        }

        while (j >= 0)
        {
            nums1[k--] = nums2[j--];
        }
    }
}

File: pythonCode/Day1.py
class Solution:
    def merge(self, nums1, m, nums2, n):
        nums1 = [x for x in nums1 if x != 0]
        nums2 = [x for x in nums2 if x != 0]
        if n == 0:
            return nums1
        elif m == 0:
            return nums2
    
        # if m < m + n:
        #     for i in range(m, m + n):
        #         nums1.append(0)
        
        for N in range(len(nums2)):
            for M in range(len(nums1)):           
                if nums1[M] > nums2[N]:
                    nums1.insert(M,nums2[N])
                    break
                if max(nums1) <= nums2[N]:
                    nums1.append(nums2[N])
                    break

        return nums1



s = Solution()
nums1 = [1,2,3,7,8]
nums2 = [1,2,5,6,6]
m = 5
n = 5
r = s.merge(nums1,m,nums2,n)
print(r)
